applications/API/views.py

```
@@ -0,0 +1,655 @@
+            queryset = queryset.filter(nsfw=nsfw)
+        # Filter by category if provided
+
+        if category := request.query_params.get("category"):
+            queryset = queryset.filter(category=category)
+
+        if not queryset.exists():
+            return Response(
+                {"detail": "No insults found matching the criteria."}, status=404
+            )
+
+        random_insult = random.choice(queryset)
```

Ruff (0.12.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)